### ISSUE TITLE
Clippy lint `clippy::str_to_string` fails on generated commands with `String` arguments

### ISSUE THREAD

**Issue by zedseven** [kangalioo/poise#39]
I run a fairly extensive set of lints on my projects. Since updating Poise from 2746481a6cef9c141ac7940b5f1eed240dff33a8 to 60106518ea0ec5940682a724c257b246ff8ac2b2, I've found  that the lint `clippy::str_to_string` warns for commands that have `String` arguments.

```rust
#![warn(clippy::str_to_string)]

#[command(prefix_command)]
pub async fn some_command(
	ctx: PoiseContext<'_>,
	#[rest]
	#[description = "Some description."]
	command: String,
) -> Result<(), Error> {
	Ok(())
}
```

Warns with the following error:
```
warning: `to_string()` called on a `&str`
  --> src\commands\mod.rs:40:1
   |
40 | #[command(prefix_command)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider using `.to_owned()`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#str_to_string
   = note: this warning originates in the macro `$crate::_parse_prefix` (in Nightly builds, run with -Z macro-backtrace for more info)
```

It seems to only be with `prefix_command`, not `slash_command`. Not a huge deal at all, but I thought I'd mention it.

**Comment by kangalioo** [comment]
Thanks for the report

Fixed in 7830e76010621ca8a380ef537534ae1ed9b725e9

If you don't mind, can you give me the list of clippy lints you use? I'll put them into the example code to make sure the generated code doesn't trigger them again

**Comment by zedseven** [comment]
Sure, here's the list I'm running: https://github.com/zedseven/radium/blob/dev/src/main.rs#L4-L38

Keep in mind these are partly personal preference and there's quite a few, so you don't have to set them up.
